File: backend/app/schemas/private_traffic.py
"""私域流量 Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SubscriptionLinkCreate(BaseModel):
    platform: str
    url: str
    subscriber_count: int = 0
    monthly_revenue: float = 0
    currency: str = "CNY"


class SubscriptionLinkResponse(BaseModel):
    id: str
    user_id: str
    platform: str
    url: str
    subscriber_count: int
    monthly_revenue: float
    currency: str
    is_active: bool
    created_at: str


class FanCommunityCreate(BaseModel):
    platform: str
    name: str
    invite_url: Optional[str] = None
    member_count: int = 0
    tags: Optional[list[str]] = None
    description: Optional[str] = None


class FanCommunityResponse(BaseModel):
    id: str
    user_id: str
    platform: str
    name: str
    invite_url: Optional[str] = None
    member_count: int
    tags: Optional[list[str]] = None
    description: Optional[str] = None
    is_active: bool


class FunnelEntryCreate(BaseModel):
    source_platform: str
    public_views: int = 0
    profile_clicks: int = 0
    link_clicks: int = 0
    converted_subscribers: int = 0
    notes: Optional[str] = None


class FunnelSummary(BaseModel):
    total_public_views: int
    total_profile_clicks: int
    total_link_clicks: int
    total_converted: int
    overall_conversion_rate: float
    by_platform: list[dict]

File: frontend-web/src/types/privateTraffic.ts
/** 私域流量类型定义 */

export interface SubscriptionLink {
  id: string
  platform: string
  url: string
  subscriber_count: number
  monthly_revenue: number
  currency: string
  total_monthly: number
}

export interface FanCommunity {
  id: string
  platform: string
  name: string
  invite_url?: string
  member_count: number
  tags?: string[]
  description?: string
}

export interface FunnelSummary {
  total_public_views: number
  total_profile_clicks: number
  total_link_clicks: number
  total_converted: number
  overall_conversion_rate: number
  by_platform: Array<{
    platform: string
    views: number
    clicks: number
    links: number
    converted: number
    profile_ctr: number
    link_ctr: number
    conv_rate: number
  }>
}
